packages/shared/src/components/sidebar/ContributeSection.tsx

```diff
@@ -46,39 +31,38 @@ export function ContributeSection({
       ),
       title: 'Submit article',
       action: trackAndShowSubmitArticle,
-      active: showSubmitArticle,
+      active: modal.type === LazyModal.SubmitArticle,
     },
     {
       icon: (active) => (
         <ListIcon Icon={() => <EmbedIcon secondary={active} />} />
       ),
       title: 'Suggest new source',
-      action: () => setShowNewSourceModal(true),
-      active: showNewSourceModal,
+      action: () => openModal({ type: LazyModal.NewSource }),
+      active: modal.type === LazyModal.NewSource,
     },
   ];
 
+  useEffect(() => {
```

Comment:
Since we are now using the lazy modal for the modals here, the hook that uses this side effect became so much simpler now and have decided to just put it here.

Referring to the `useSubmitArticle` hook.

Reply:
I would still put the tracking inside the modal actually.
Maybe the state that evaluates if we show the modal, but I'm unsure if it's actually used anymore.

Reply:
I have now introduced the user state for identifying whether the user is eligible in the boot endpoint. So from that, it would result the same value.

Reply:
I'm not 100% sure I follow on this part, maybe it's best we sync on this quickly.

Reply:
Cool. Let me know once you are free, I should be too.

Reply:
This is basically the useEffect inside the deleted hook. This was never inside the modal.